### PR TITLE
Feature Slicing

### PR DESCRIPTION
**Pull request by thomasborgen**

### CODE REVIEW

File: docs/introduction.md [hunk 1]
```diff
@@ -381,6 +381,156 @@ To find more values and combine them, simply add another `mapping` object to `ma
...
+            {
+                "name": "unicorn",
+                "street": "Fluffy St. 40"
+            }
+        ]
+    }
+    ```
+
+!!! Hint
+    If you have some max length on a database table, then you can use string slicing to make sure the length does not exceed a certain length with the `to` key. Some databases also has for example two address fields for when the length of one is too short. Then map both with slicing [from=0 to=50] and from=50 to=None respectively and you'll solve the problem.
```
Comment (ChameleonTartu) [inline docs/introduction.md:455]: `to=None` or `to=null`?

File: docs/introduction.md [hunk 2]
```diff
@@ -381,6 +381,156 @@ To find more values and combine them, simply add another `mapping` object to `ma
...
+            {
+                "name": "unicorn",
+                "birthday": "1991-12-30"
+            }
+        ]
+    }
+    ```
+
+!!! Hint
+    If you need to take values from end of string, like the 5 last characters, then you can use a negative `from` value to count from the end instead. This works just like pythons slicing functionality.
```
Comment (ChameleonTartu) [inline docs/introduction.md:531]: Wouldn't be a link to Python slicing appropriate here?

File: tests/schema/test_slicing.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from jsonschema import ValidationError
+from returns.pipeline import is_successful
```
Comment (ChameleonTartu) [inline tests/schema/test_slicing.py:2]: It would be nice to have a single test with negative numbers so that slicing will reverse string.

**Review by ChameleonTartu — CHANGES_REQUESTED** [review]
@thomasborgen LGTM. There are minor comments regarding the functionality.

File: tests/schema/test_slicing.py (hunk 3 above)
Comment (thomasborgen) [reply tests/schema/test_slicing.py:2]: Slicing cannot reverse the string. consider:
```python
>>> 'abcdefg'[-3:-5]
''
```
slicing notation actually has one more parameter and that is the `step`. By default its 1, but changing it to -1 reversers the order. so:
```python
>>> 'abcdefg'[-3:-5:-1]
'ed'
```
We can add an issue about this if we need this functionality, but i believe it to be very very nishe. it would also allow you to skip values ie:
```python
>>> 'abcdefg'[::2]
'aceg'
```

I'll add tests with negative values.

**Review by ChameleonTartu — APPROVED** [review]
@thomasborgen パーフェクト!
